### PR TITLE
doc: Add Python 3.10 page.

### PR DESCRIPTION
**Pull request by dlech**
This adds a new page for Python 3.10 implementation status similar to previous releases.

### CODE REVIEW

File: docs/differences/python_310.rst [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+  | The hashlib module has preliminary support for OpenSSL 3.0.0.                                                 |               |
+  +---------------------------------------------------------------------------------------------------------------+---------------+
+  | The pure-Python fallback of :func:`~hashlib.pbkdf2_hmac` is deprecated. In                                    |               |
+  | the future PBKDF2-HMAC will only be available when Python has been built with                                 |               |
+  | OpenSSL support.                                                                                              |               |
+  +---------------------------------------------------------------------------------------------------------------+---------------+
+  | `os <https://docs.python.org/3/whatsnew/3.10.html#os>`_                                                                       |
+  +---------------------------------------------------------------------------------------------------------------+---------------+
+  | Add :func:`os.cpu_count()` support for VxWorks RTOS.                                                          |               |
```
Comment (mattytrentini) [inline docs/differences/python_310.rst:165]: I think we can remove this one!

File: docs/differences/python_310.rst [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+  +---------------------------------------------------------------------------------------------------------------+---------------+
+  | Add a new function :func:`os.eventfd` and related helpers to wrap the                                         |               |
```
Comment (mattytrentini) [inline docs/differences/python_310.rst:167]: Not applicable

File: docs/differences/python_310.rst [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+  | ``eventfd2`` syscall on Linux.                                                                                |               |
+  +---------------------------------------------------------------------------------------------------------------+---------------+
+  | Add :func:`os.splice()` that allows to move data between two file                                             |               |
```
Comment (mattytrentini) [inline docs/differences/python_310.rst:170]: Not applicable

File: docs/differences/python_310.rst [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+  | descriptors without copying between kernel address space and user                                             |               |
+  | address space, where one of the file descriptors must refer to a                                              |               |
+  | pipe.                                                                                                         |               |
+  +---------------------------------------------------------------------------------------------------------------+---------------+
+  | Add :data:`~os.O_EVTONLY`, :data:`~os.O_FSYNC`, :data:`~os.O_SYMLINK`                                         |               |
```
Comment (mattytrentini) [inline docs/differences/python_310.rst:175]: Not applicable

**Comment by mattytrentini** [comment]
Looks good to me; I've only raised minor points. Nice work!

File: docs/differences/python_310.rst (hunk 1 above)
Comment (dlech) [reply docs/differences/python_310.rst:165]: I just copied everything for completeness. I didn't go back an look if the previous versions had trimmed down lists for "not relevant" items or marked them as such.

Comment (dpgeorge) [reply docs/differences/python_310.rst:165]: This (and the ones below) could eventually become relevant for MicroPython, maybe.... so perhaps just leave them here for completeness.

File: docs/differences/python_310.rst [hunk 5]
```diff
@@ -0,0 +1,238 @@
+.. _python_310:
+
+Python 3.10
+===========
+
+Python 3.10.0 (final) was released on the 4 October 2021. The Features for 3.10 are
+defined in `PEP 619 <https://www.python.org/dev/peps/pep-0619/#features-for-3-10>`_
+and a detailed description of the changes can be found in
+`What's New in Python 3.10 <https://docs.python.org/3/whatsnew/3.10.html>`_
```
Comment (laurensvalk) [inline docs/differences/python_310.rst:9]: There's a period missing at the end. It's also missing on the 3.9 page but the 3.8 page has it.

**Comment by dlech** [comment]
rebased and fixed a few `.`s

**Comment by dpgeorge** [comment]
Thanks!

**Pull request merged**